src/qris_toolbar.py

```diff
@@ -313,10 +324,60 @@ def onProjectLoad(self):
         """Slot for QGIS signal when a QGIS project (QGZ) is being loaded.
         Check if the QGIS project references a QRiS project. Load it if does."""
 
-        qris_project_path, type_conversion_ok = QgsProject.instance().readEntry(CONSTANTS['settingsCategory'], CONSTANTS['qris_project_path'])
-        if type_conversion_ok is True:
-            if qris_project_path is not None and len(qris_project_path) > 0 and os.path.exists(qris_project_path):
-                self.open_qris_project(qris_project_path)
+        qris_project_path = self.get_project_path_settings()
+        if qris_project_path is not None and len(qris_project_path) > 0 and os.path.exists(qris_project_path):
+            self.open_qris_project(qris_project_path)
+
+     
+    def project_homePathChanged(self):
+        """Trigger an event before saving the project so we have an opportunity to corrent the paths
+        """
+        proj_path = self.get_project_path_settings()
+        self.set_project_path_settings(proj_path)
```

Comment:
Every time the qgz home directory changes (including from nothing to a real directory) we write and adjust the path. Relative paths are used when appropriate.